Modules/_ctypes/_ctypes.c

```diff
@@ -2613,12 +2648,11 @@ static int PyCData_NewGetBuffer(PyObject *myself, Py_buffer *view, int flags)
     view->format = dict->format ? dict->format : "B";
     view->ndim = dict->ndim;
     view->shape = dict->shape;
-    view->itemsize = self->b_size;
-    if (view->itemsize) {
-        for (i = 0; i < view->ndim; ++i) {
-            view->itemsize /= dict->shape[i];
-        }
+    item_dict = PyCData_item_stgdict(dict);
```

Comment:
Would it be possible to have `dict->itemsize` on every dict object instead of traversing the tree to find it?

Reply:
Seems reasonable since we already do that for shape. I did it this way because it seemed like a smaller change. I'll have a go at doing it the way you suggest.